testing/chain.go

```diff
@@ -305,7 +315,8 @@ func (chain *TestChain) NextBlock() {
 		ProposerAddress:    chain.CurrentHeader.ProposerAddress,
 	}
 
-	chain.App.BeginBlock(abci.RequestBeginBlock{Header: chain.CurrentHeader})
+	_, err = chain.App.FinalizeBlock(&abci.RequestFinalizeBlock{Height: chain.CurrentHeader.Height})
```

Comment:
Thanks very much!

Reply:
thankyou sir, i believed we have try to add time when call `FinalizeBlock`. Gonna check on the `SignAndDeliver`

Reply:
seem that the time in the ctx is behind that in the current chain header

Reply:
fyi: I have worked around most of the issues in the wasmd [fork](https://github.com/CosmWasm/wasmd/tree/sdk-upgrade050/x/wasm/ibctesting) of Testchain. Please 👀 for some inspiration. I am blocked on proof verification, though.

Reply:
thank you @alpe !

Reply:
Hi @alpe, thanks for the fixes! I looked at your latest changes and the proof verification failure. It looks like the proofs are passing, but verification of the consensus state is failing. This is because the historical info set in the staking module seems to have an empty root hash and next validator hash, I'm not sure why. It might have to do with this being block height 2 at the time of execution. See [here](https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/keeper/historical_info.go#L127), other header fields seem to be filled in. 

Looks like historical info in staking needs to be updated to use the non-deprecated field? https://github.com/cosmos/cosmos-sdk/blob/main/types/context.go#L43 cc @tac0turtle 

I think it should maybe be using `sdkCtx.CometInfo.GetValidatorHash()` for next validator hash and `sdkCtx.HeaderInfo().GetHash()` for the app hash

Reply:
However, it does look like the header is being [set](https://github.com/cosmos/cosmos-sdk/blob/main/types/context.go#L130) in context, is it expected to be empty?

Reply:
Ah! The testing package will need to pass in those arguments to the `FinalizeBlock` request